src/spdx.lalrpop

```diff
@@ -0,0 +1,60 @@
+use lalrpop_util::ParseError;
+use constants::{DOCUMENT_REFERENCE, EXCEPTIONS, LICENSES};
+use regex::Regex;
+
+grammar;
+
+extern {
+    type Location = usize;
+    type Error = &'input str;
+}
+
+pub Expr: () = {
```

Comment:
Does the SPDX spec say that `License OR License AND License` is valid without parens? (this grammar makes that valid) Does it say `License AND License OR License` is not valid without parens? (this grammar makes that invalid).

If not, would it be simpler and more correct to define this like:

```
Expr = Term AND Term | Term OR Term | Term

Term = Id WithClause? | ( Expr )
```

Reply:
> Does the SPDX spec say that License OR License AND License is valid without parens? (this grammar makes that valid) Does it say License AND License OR License is not valid without parens? (this grammar makes that invalid).

The SPDX 2.1 spec [defines operator precedence][1], and `AND` is higher than `OR` (as [in C][2] and likely lots of other syntaxes).  So both of your examples are valid and well-defined.

[1]: https://spdx.org/spdx-specification-21-web-version#h.jxpfx0ykyb60
[2]: https://en.wikipedia.org/wiki/Operators_in_C_and_C%2B%2B#Operator_precedence